test/extended/router/metrics.go

```diff
@@ -58,13 +58,17 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router] openshift rou
 			}
 		}
 
-		epts, err := oc.AdminKubeClient().CoreV1().Endpoints("default").Get("router", metav1.GetOptions{})
+		// wait for the router endpoints to show up
+		err = wait.PollImmediate(2*time.Second, 120*time.Second, func() (bool, error) {
+			epts, err := oc.AdminKubeClient().CoreV1().Endpoints("default").Get("router", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
Don't you want to retry if the endpoint does not exist yet?

Reply:
Odds of that happening are so low as to be not worth it.  If the endpoints controller has taken 2-3 minutes to create the endpoint we're broke.

Reply:
Or the deployer pod has yet to start.

Reply:
No, endpoints object is created immediately.  It's just empty.